### DISCUSSION TITLE
Wildcard behavior is different from bash

### DISCUSSION THREAD

**Discussion by Nikaidou-Shinku** [nushell/nushell#7917] [Q&A]
Nushell's globs really confuse me. I noticed today that sometimes Nushell handles them in a very different way than bash.

As an example, bash will always try to match shell patterns, and if it doesn't match anything, it will pass the patterns to the program as a literal.
When I run `touch *` in bash, it will consider the `*` as a wildcard, and only when it can not match anything bash passed `*` as a literal to `touch` and created a file named `*`.

However, things are very different at Nushell's built-in command.
When I run `touch *` in Nushell, it will always create a file named `*` whether it can match anything or not.

There are many other cases where they behave differently.
For example, when I quote the wildcard, bash will no longer try to match anything. Run `rm '*'` in bash will delete only the file named `*`. But Nushell always deletes all files.

When I use `^touch` or `^rm`, Nushell behaves the same as bash. Maybe the key problem is the built-in command? Is this difference expected?

I searched the discussions and found #6066, but it doesn't seem to help much.

**Comment by fdncred** [comment]
I agree, glob handling isn't real good. One of the problems, IMO, is that we try to be too much like other shells. For instance, a true nushell way of doing things like you suggest would be to enable something like `glob * | each {|r| $r | touch}`. So, this type of functionality, narrows all the globbing functionality to 1 command which is glob. The way it is now, every command like `touch` has to have its own globbing. Granted, that could be a shared function, but you still have to program every command to respond to it in order to be consistent.

To top it off, we have nu-glob and the wax crate that provide different ways of globbing. wax is used in the glob command and nu-glob, which is a modified glob crate, handles globbing in other areas, like `ls`.

We're open to suggestions and PRs. Thanks for the comments.
